modules/Microsoft.VirtualMachineImages/imageTemplates/.test/advanced/deploy.test.bicep

```diff
@@ -0,0 +1,105 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+param resourceGroupName string = 'ms.virtualmachineimages.imagetemplates-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to.')
+param location string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'vmitadv'
+
+@description('Optional. The version of the Azure Compute Gallery Image Definition to be added.')
+param sigImageVersion string = utcNow('yyyy.MM.dd')
+
+@description('Optional. The staging resource group name in the same location and subscription as the image template. Must not exist.')
+param stagingResourceGroupName string = 'ms.virtualmachineimages.imagetemplates-${serviceShort}-staging-rg'
+
+@description('Optional. Enable telemetry via a Globally Unique Identifier (GUID).')
+param enableDefaultTelemetry bool = true
+
+// =========== //
+// Deployments //
+// =========== //
+
+// General resources
+// =================
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+  name: resourceGroupName
+  location: location
+}
+
+module resourceGroupResources 'dependencies.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(resourceGroupName, location)}-paramNested'
+  params: {
+    managedIdentityName: 'dep-<<namePrefix>>-msi-${serviceShort}'
+    sigImageDefinitionName: 'dep-<<namePrefix>>-imgd-${serviceShort}'
+    galleryName: 'dep<<namePrefix>>sig${serviceShort}'
+    virtualNetworkName: 'dep<<namePrefix>>-vnet-${serviceShort}'
+  }
+}
+
+// required for the Azure Image Builder service to assign the list of User Assigned Identities to the Build VM.
+resource msi_managedIdentityOperatorRoleAssignment 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
```

Comment:
Same question as for the `common` test: How is it that it does not need Contributor permissions to create the Staging Resources?

Reply:
In this case the managed identity assigned at the vm image builder service level needs the ability to assign user-assigned identities to the build VM.